packages/block-editor/src/store/private-selectors.js

```diff
@@ -8,3 +28,88 @@
 export function isBlockInterfaceHidden( state ) {
 	return state.isBlockInterfaceHidden;
 }
+
+/**
+ * Gets the client ids of the last inserted blocks.
+ *
+ * @param {Object} state Global application state.
+ * @return {Array|undefined} Client Ids of the last inserted block(s).
+ */
+export function getLastInsertedBlocksClientIds( state ) {
+	return state?.lastBlockInserted?.clientIds;
+}
+
+/**
+ * @typedef {import('../components/block-editing-mode').BlockEditingMode} BlockEditingMode
+ */
+
+/**
+ * Returns the block editing mode for a given block.
+ *
+ * The mode can be one of three options:
+ *
+ * - `'disabled'`: Prevents editing the block entirely, i.e. it cannot be
+ *   selected.
+ * - `'contentOnly'`: Hides all non-content UI, e.g. auxiliary controls in the
+ *   toolbar, the block movers, block settings.
+ * - `'default'`: Allows editing the block as normal.
+ *
+ * Blocks can set a mode using the `useBlockEditingMode` hook.
+ *
+ * The mode is inherited by all of the block's inner blocks, unless they have
+ * their own mode.
+ *
+ * A template lock can also set a mode. If the template lock is `'contentOnly'`,
+ * the block's mode is overridden to `'contentOnly'` if the block has a content
+ * role attribute, or `'disabled'` otherwise.
+ *
+ * @see useBlockEditingMode
+ *
+ * @param {Object} state    Global application state.
+ * @param {string} clientId The block client ID, or `''` for the root container.
+ *
+ * @return {BlockEditingMode} The block editing mode. One of `'disabled'`,
+ *                            `'contentOnly'`, or `'default'`.
+ */
+export const getBlockEditingMode = createRegistrySelector(
+	( select ) =>
+		( state, clientId = '' ) => {
```

Comment:
You [mentioned](https://github.com/WordPress/gutenberg/pull/50643#issuecomment-1554060494) this selector might need to be memoized (?)

Reply:
Yes!

Unfortunately it's not possible to memoize a `createRegistrySelector` selector, but that's okay, because there are only two selectors that `getBlockEditingMode` calls that do a decent amount of computation (read: they have a `for` loop), and both of these are memoized. They are `getExplicitBlockEditingMode` and `__experimentalHasContentRoleAttribute`. In other words `getBlockEditingMode` is O(1) when called a second time with the same arguments.